modules/kargoBidAdapter.js

```diff
@@ -0,0 +1,166 @@
+import * as utils from 'src/utils';
+import {config} from 'src/config';
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'kargo';
+const HOST = 'https://krk.kargo.com';
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [],
+  isBidRequestValid: function(bid) {
+    if (!bid || !bid.params) {
+      return false;
+    }
+    return !!bid.params.placementId;
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const currency = config.getConfig('currency');
+    const transformedParams = Object.assign({}, {
+      timeout: bidderRequest.timeout,
+      currency: currency,
+      cpmGranularity: 1,
+      cpmRange: {
+        floor: 0,
+        ceil: 20
+      },
+      adSlotIds: utils._map(validBidRequests, bid => bid.params.placementId)
+    }, spec._getAllMetadata());
+    const encodedParams = encodeURIComponent(JSON.stringify(transformedParams));
+    return Object.assign({}, bidderRequest, {
+      method: 'GET',
+      url: `${HOST}/api/v1/bid`,
+      data: `json=${encodedParams}`,
+      currency: currency
+    });
+  },
+  interpretResponse: function(adUnits, bidRequest) {
+    let bids = {};
+    utils._each(bidRequest.bids, bid => bids[bid.params.placementId] = bid);
+    const bidResponses = [];
+    for (let adUnitId in adUnits) {
+      let adUnit = adUnits[adUnitId];
+      bidResponses.push({
+        requestId: bids[adUnitId].bidId,
+        bidderCode: spec.code,
+        cpm: Number(adUnit.cpm),
+        width: adUnit.width,
+        height: adUnit.height,
+        ad: adUnit.adm,
+        ttl: 300,
+        creativeId: adUnitId,
+        netRevenue: true,
+        currency: bidRequest.currency
+      });
+
+      if (adUnit.receivedTracker) {
+        let el = document.createElement('img');
+        el.src = adUnit.receivedTracker;
+        document.body.appendChild(el);
+      }
+    }
+    return bidResponses;
+  },
+
+  // PRIVATE
+  _readCookie(name) {
+    let nameEquals = `${name}=`;
+    let cookies = document.cookie.split(';');
+
+    for (let key in cookies) {
+      let cookie = cookies[key];
+      while (cookie.charAt(0) === ' ') {
+        cookie = cookie.substring(1, cookie.length);
+      }
+
+      if (cookie.indexOf(nameEquals) === 0) {
+        return cookie.substring(nameEquals.length, cookie.length);
+      }
+    }
+
+    return null;
+  },
+
+  _getCrbIds() {
+    try {
+      const crb = JSON.parse(decodeURIComponent(spec._readCookie('krg_crb')));
```

Comment:
what's this cookie you are reading? How do you have access to it if you are in the pub domain?

Reply:
These are first party cookies our pubs are likely to already have on their page from our various other integrations.

Reply:
ok that's what I thought. Thanks for clarifying.

Reply:
No problem